storage/table_entities.go

```diff
@@ -0,0 +1,379 @@
+package storage
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"reflect"
+)
+
+const (
+	partitionKeyNode                    = "PartitionKey"
+	rowKeyNode                          = "RowKey"
+	tag                                 = "table"
+	tagIgnore                           = "-"
+	continuationTokenPartitionKeyHeader = "X-Ms-Continuation-Nextpartitionkey"
+	continuationTokenRowHeader          = "X-Ms-Continuation-Nextrowkey"
+	maxTopParameter                     = 1000
+)
+
+type queryTablesResponse struct {
+	TableName []struct {
+		TableName string `json:"TableName"`
+	} `json:"value"`
+}
+
+// TableEntity interface specifies
+// the functions needed to support
+// marshaling and unmarshaling into
+// Azure Tables. The struct must only contain
+// simple types because Azure Tables do not
+// support hierarchy.
+type TableEntity interface {
+	PartitionKey() string
+	RowKey() string
+	SetPartitionKey(string) error
```

Comment:
The idea here was to give the implementer the chance of aborting the method if needed (for example if the format is not valid).

Referencing the above example: 

```go
func (c *MyType) PartitionKey() string {
    return "primarykey"
}

func (c *MyType) RowKey() string {
    return c.Surname + " " + c.Name // <-- translate from user domain to data domain
}

func (c *MyType) SetPartitionKey(s string) error {
    return nil // nothing to do, PartitionKey is hardcoded and data domain only
}

func (c *MyType) SetRowKey(s string) error {
  f := string.Fields(s) <-- translate from data domain to user domain
  c.Surname = f[0]
  c.Name= f[1]
  return nil
}
```
It's clear how the ```SetRowKey``` method could fail (perhaps only one field was provided?). In the code there is no check but you could (should, I was lazy :innocent:) prevent the panic and return a nicer error instead.

I agree most implementations won't benefit from it though. At any rate in case is not needed it's just a ```return nil``` more in the code. What do you think?